### ISSUE TITLE
What is the suitable encoding of wav file?

### ISSUE THREAD

**Issue by k5-inoue** [sarulab-speech/xvector_jtubespeech#5]
The output value of `scipy.io.wavfile.read` depends on WAV format. (https://docs.scipy.org/doc/scipy/reference/generated/scipy.io.wavfile.read.html)
I think the WAV format is necessary information for extracting suitable x-vector.
Can I understand `sample.wav` has a suitable encoding for this script?

* WAV format: pcm_s16le (**16-bit integer PCM**) <- I wanna know here!
* Sampling rate: 16000 Hz
* Channel: 1 (mono)

**Comment by vebmaylrie** [comment]
> WAV format: pcm_s16le (16-bit integer PCM) <- I wanna know here!
yes, but another format, e.g., 24-bit PCM, is also acceptable. This is because the audio data is converted into the float format.

**Comment by k5-inoue** [comment]
>This is because the audio data is converted into the float format.

I think `astype(np.float32)` is only cast into the `np.float32` in the Usage code.
The audio array value depends on the WAV format because of `scipy.io.wavfile.read`.
I tried the next sample code showing the audio array value in some WAV formats.


```
# convert WAV format
import torchaudio
audio_norm, audio_sampling_rate = torchaudio.load("sample.wav")
torchaudio.save("sample_16bit_PCM.wav", audio_norm, 16000, encoding="PCM_S", bits_per_sample=16)
torchaudio.save("sample_24bit_PCM.wav", audio_norm, 16000, encoding="PCM_S", bits_per_sample=24)
torchaudio.save("sample_32bit_PCM.wav", audio_norm, 16000, encoding="PCM_S", bits_per_sample=32)
torchaudio.save("sample_32bit_float.wav", audio_norm, 16000, encoding="PCM_F", bits_per_sample=32)

# read with scipy
from scipy.io import wavfile
_, wav = wavfile.read("sample.wav")
_, wav_16pcm = wavfile.read("sample_16bit_PCM.wav")
_, wav_24pcm = wavfile.read("sample_24bit_PCM.wav")
_, wav_32pcm = wavfile.read("sample_32bit_PCM.wav")
_, wav_32float = wavfile.read("sample_32bit_float.wav")

# cast with astype
import numpy as np
casted_wav_16pcm = wav_16pcm.astype(np.float32)
casted_wav_24pcm = wav_24pcm.astype(np.float32)
casted_wav_32pcm = wav_32pcm.astype(np.float32)
casted_wav_32float = wav_32float.astype(np.float32)
```

The values of each array are shown below.

```
# read with scipy
wav_16pcm: array([ 6,  9,  8, ..., 18, 19, 20], dtype=int16)
wav_24pcm: array([ 393216,  589824,  524288, ..., 1179648, 1245184, 1310720], dtype=int32)
wav_32pcm: array([ 393216,  589824,  524288, ..., 1179648, 1245184, 1310720], dtype=int32)
wav_32float: array([0.00018311, 0.00027466, 0.00024414, ..., 0.00054932, 0.00057983, 0.00061035], dtype=float32)

# cast with astype
casted_wav_16pcm : array([ 6,  9,  8, ..., 18, 19, 20], dtype=float32)
casted_wav_24pcm: array([ 393216,  589824,  524288, ..., 1179648, 1245184, 1310720], dtype=float32)
casted_wav_32pcm: array([ 393216,  589824,  524288, ..., 1179648, 1245184, 1310720], dtype=float32)
casted_wav_32float: array([0.00018311, 0.00027466, 0.00024414, ..., 0.00054932, 0.00057983, 0.00061035], dtype=float32)
```

If `kaldi.mfcc` or `model.vectorize` did normalize into the range -1 ~ +1, there is no ploblem.
In light of these factors, what do you think?

**Comment by vebmaylrie** [comment]
Sorry that I misunderstood. Please use 16bit PCM.

**Comment by k5-inoue** [comment]
Thanks!!!
